src/Features/CSharp/Portable/InlineHints/CSharpInlineTypeHintsService.cs

```diff
@@ -80,6 +80,15 @@ public CSharpInlineTypeHintsService()
                 }
             }
 
+            if (node is ImplicitObjectCreationExpressionSyntax { Parent: ArgumentSyntax } implicitNew)
+            {
+                var type = semanticModel.GetTypeInfo(implicitNew, cancellationToken).Type;
+                if (IsValidType(type))
+                {
+                    return (type, new TextSpan(implicitNew.NewKeyword.Span.End, 0));
```

Comment:
The space needs to go before the type:
```CS
M(new())
// Expected:
M(p: new TypeName())
// Actual:
M(p: newTypeName ())
```
![d9EUzETks2](https://user-images.githubusercontent.com/24472128/103934994-37b2bd80-5126-11eb-99af-b09600da5abf.gif)

The root cause is here:

https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/InlineHints/AbstractInlineTypeHintsService.cs#L76-L78

I need more flexibility in the return type of `(ITypeSymbol type, TextSpan span)? CSharpInlineTypeHintsService.TryGetTypeHint(...)`. Do you have any suggestions on how this could be achieved best?

Reply:
there are a few options i can think of.  

1. just allow more information to be passed back in InlineHint relating to: preceding/following whitespace.  
2. don't do '1', but have this code add the preceding whitespace if this would abut some previous non-whitespace text.

Reply:
I went with option 1 (see 125366d7)